geode-core/src/main/java/org/apache/geode/cache/query/internal/CqEntry.java

```diff
@@ -82,6 +82,9 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object other) {
+    if (!(other instanceof CqEntry)) {
+      return false;
+    }
     if (other == null) {
```

Comment:
I think this if check is no longer needed with the addition of the instanceof check

Reply:
my bad, I missed that!!